zowex/src/proc.rs

```diff
@@ -269,22 +271,37 @@ fn read_pid_for_user() -> Option<sysinfo::Pid> {
  * On Linux:
  *      If you use Stdio::inherit(), you get double output. If use use Stdio::null(), you get no color.
  */
+
 pub fn proc_start_daemon(njs_zowe_path: &str) -> String {
     println!("Starting a background process to increase performance ...");
 
-    if env::consts::OS == "windows" {
-        /* Windows CMD and Powershell terminal windows show escape characters
-         * instead of colors in daemon-mode. A more elegant solution may exist,
-         * but for now we just turn off color in daemon mode on Windows.
-         */
-        env::set_var("FORCE_COLOR", "0");
-    }
-
     let daemon_arg = LAUNCH_DAEMON_OPTION;
+    #[cfg(not(target_family = "windows"))]
+    match Command::new(njs_zowe_path)
+        .arg(daemon_arg)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()
+    {
+        Ok(_unused) => { /* nothing to do */ }
+        Err(error) => {
+            println!(
+                "Failed to start the following process:\n    {} {}",
+                njs_zowe_path, daemon_arg
+            );
+            println!("Due to this error:\n    {}", error);
+            std::process::exit(EXIT_CODE_CANNOT_START_DAEMON);
+        }
+    };
+
+    // Uses creation flags from https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
+    // Flags are CREATE_NO_WINDOW, CREATE_NEW_PROCESS_GROUP, and CREATE_UNICODE_ENVIRONMENT
+    #[cfg(target_family = "windows")]
     match Command::new(njs_zowe_path)
         .arg(daemon_arg)
         .stdout(Stdio::null())
         .stderr(Stdio::null())
+        .creation_flags(0x08000600)
         .spawn()
```

Comment:
Would something like this work to consolidate the code?
```rust
    let mut command = Command::new(njs_zowe_path)
        .arg(daemon_arg)
        .stdout(Stdio::null())
        .stderr(Stdio::null());
    #[cfg(target_family = "windows")]
    command.creation_flags(0x08000600);

    command.spawn()
    {
        Ok(_unused) => { /* nothing to do */ }
        Err(error) => {
            println!(
                "Failed to start the following process:\n    {} {}",
                njs_zowe_path, daemon_arg
            );
            println!("Due to this error:\n    {}", error);
            std::process::exit(EXIT_CODE_CANNOT_START_DAEMON);
        }
    };
```

Reply:
I cannot do this, because it would involve reassigning `command`:

`command = command.creation_flags(0x08000600)`, and attributes on expressions such as that are not currently supported in rust, and are experimental at this time.

Reply:
I believe the fact that `command.creation_flags` returns a command object is purely for convenience, so that multiple `command.XXX` functions can be chained together. The original command object gets mutated so it's not required to reassign the command variable to a new value.

Also there were a few mistakes in the code snippet I posted - sorry about that 😢 Updated the snippet in my above comment and it built for me locally without errors.

Reply:
I did test this yesterday with @traeok, and spawning it like this still caused it to be spawned as a subprocess of the current terminal.

Reply:
I thought we had tried it without reassigning command, and not sure why it wasn't working but as @awharn mentioned the process was still spawning as a subprocess of the terminal.

But, we can change the argument for `creation_flags` to `0x08000200` as any argument passed here will alway's be OR'ed with `CREATE_UNICODE_ENVIRONMENT`:
https://doc.rust-lang.org/stable/std/os/windows/process/trait.CommandExt.html#tymethod.creation_flags

Reply:
I don't know what changed between yesterday and today, but it seems to work now. 🤷